includes/Core/Nonces/Nonces.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Nonces\Nonces
+ *
+ * @package   Google\Site_Kit\Core\Nonces
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Nonces;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Authentication\Authentication;
+use Google\Site_Kit\Core\Permissions\Permissions;
+use Google\Site_Kit\Core\REST_API\REST_Route;
+use Google\Site_Kit\Core\REST_API\REST_Routes;
+use Google\Site_Kit\Core\Util\Feature_Flags;
+use WP_REST_Response;
+use WP_REST_Server;
+
+/**
+ * Class managing plugin permissions.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+final class Nonces {
+	/*
+	 * Nonce actions
+	 *
+	 * @since n.e.x.t
+	 */
+	const NONCE_UPDATES = 'updates';
+
+	/**
+	 * Plugin context.
+	 *
+	 * @since n.e.x.t
+	 * @var Context
+	 */
+	private $context;
+	/**
+	 * Array of nonce actions.
+	 *
+	 * @since n.e.x.t
+	 * @var array
+	 */
+	private $nonce_actions;
+
+	/**
+	 * Constructor.
+	 *
+	 * Sets up the capability mappings.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Context $context Plugin context.
+	 */
+	public function __construct( Context $context ) {
+		$this->context = $context;
+
+		$this->nonce_actions = array(
+			self::NONCE_UPDATES,
+			'testnonce',
```

Comment:
```suggestion
			'testnonce',
```

Is this a test value being initialised in the class? I'm a bit confused about this.

Reply:
Removed this.